moo.js

```diff
@@ -140,6 +141,22 @@
     return result
   }
 
+  function isKeyword(literal, otherRules) {
+    for (var i=0; i<otherRules.length; i++) {
+      var rule = otherRules[i]
+      var match = rule.match
+      for (var j=0; j<match.length; j++) {
+        var pat = match[j]
+        if (isRegExp(pat)) {
```

Comment:
`continue` can help here with the accreting `}`s (especially if you put it on one line as ` continue`, not `{\n  continue\n}`).